src/query/storages/fuse/src/operations/mutation/compact_mutator/full_compact_mutator.rs

```diff
@@ -96,17 +93,32 @@ impl FullCompactMutator {
 #[async_trait::async_trait]
 impl TableMutator for FullCompactMutator {
     async fn target_select(&mut self) -> Result<bool> {
-        let snapshot = self.base_snapshot.clone();
+        let snapshot = self.compact_params.base_snapshot.clone();
         let segment_locations = &snapshot.segments;
-        let mut summarys = Vec::new();
 
+        // Blocks that need to be reorganized into new segments.
+        let mut remain_blocks = Vec::new();
         // Read all segments information in parallel.
         let segments_io = SegmentsIO::create(self.ctx.clone(), self.data_accessor.clone());
-        let segments = segments_io.read_segments(segment_locations).await?;
-        for (idx, segment) in segments.iter().enumerate() {
+        let segments = segments_io
+            .read_segments(segment_locations)
```

Comment:
If we have the limit, is there no need to read all segments?

Reply:
> If we have the limit, is there no need to read all segments?

No, because it needs to calculate snapshot.summary.